calico_node/startup/startup.go

```diff
@@ -904,6 +915,27 @@ func ensureDefaultConfig(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, c
 	return nil
 }
 
+// ensureMigrated ensures any data migration needed is done.
+func ensureMigrated(ctx context.Context, cfg *apiconfig.CalicoAPIConfig, cv3 client.Interface) error {
+	_, cv1, err := clients.LoadClients("", "")
+	if err != nil {
+		return err
+	}
+	m := migrator.New(cv3, cv1, nil)
+	if yes, err := m.ShouldMigrate(); err == nil {
```

Comment:
this could be made a little more succinct and with less nesting:

```
	if yes, err := m.ShouldMigrate(); err != nil {
		return err
	} else if yes {
		if _, err = m.Migrate(); err != nil {
			return errors.New(fmt.Sprintf("Migration failed: %v", err))
		}
	}
```

Reply:
Conventionally it's bad form to inline a conditional if the function returns two arguments.